### PR TITLE
Add support to tar flag for docker scan

### PR DESCRIPTION
**Pull request by guyshe-jfrog**
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [ ] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

Part of https://github.com/jfrog/jfrog-cli/pull/2451
-----

### CODE REVIEW

File: commands/scan/dockerscan.go [hunk 1]
```diff
@@ -63,17 +68,19 @@ func (dsc *DockerScanCommand) Run() (err error) {
...
-	log.Info("Creating image archive...")
-	imageTarPath := filepath.Join(tempDirPath, "image.tar")
-	dockerSaveCmd := exec.Command("docker", "save", dsc.imageTag, "-o", imageTarPath)
-	var stderr bytes.Buffer
-	dockerSaveCmd.Stderr = &stderr
-	err = dockerSaveCmd.Run()
-	if err != nil {
-		return fmt.Errorf("failed running command: '%s' with error: %s - %s", strings.Join(dockerSaveCmd.Args, " "), err.Error(), stderr.String())
+	imageTarPath := dsc.imageTag
+	if !dsc.isTar {
```
Comment (hadarshjfrog) [inline commands/scan/dockerscan.go:72]: shouldn't we want to keep the regular flow as un-tabbed as possible?
if it is Tar - and we do... nothing? shouldn't we want to log it at least?

Comment (guyshe-jfrog) [reply commands/scan/dockerscan.go:72]: The UI prints indexing filename.tar so it is logged to the screen:
Screenshot 2024-02-25 at 21 55 21

File: commands/scan/dockerscan.go [hunk 2]
```diff
@@ -63,17 +68,19 @@ func (dsc *DockerScanCommand) Run() (err error) {
...
+		if dsc.progress != nil {
```
Comment (attiasas) [inline commands/scan/dockerscan.go:73]:
```suggestion
        skipSave := dsc.isTar || (fileutils.IsFileExists(dsc.imageTag) && strings.HasSuffix(".tar"))
		if !skipSave {
```
I would add a test to check if this given argument is an existing file path (with .tar suffix) or if this flag is provided, to force skiping.

File: commands/audit/audit.go [hunk 3]
```diff
@@ -56,6 +57,11 @@ func (auditCmd *AuditCommand) SetIncludeLicenses(include bool) *AuditCommand {
 	return auditCmd
 }
 
+func (auditCmd *AuditCommand) SetIsTar(isTar bool) *AuditCommand {
+	auditCmd.IsTar = isTar
+	return auditCmd
+}
```
Comment (attiasas) [inline commands/audit/audit.go:63]: Why you need it in audit command? this is just for docker scan....

**Review by attiasas — CHANGES_REQUESTED** [review]
1. Change the description and name of the Argument, now that it is not only a tag (at cli/docs/scan/dockerscan/help.go)

2. Add integration tests that runs scan on .tar file

**Review by guysoft — APPROVED** [review]

**Comment by attiasas** [comment]
You can use the `scan` command to scan binary

**Pull request closed without merging**